src/renderer.ts

```diff
@@ -203,16 +214,436 @@ export class DefaultCellRenderer implements ICellRenderer {
+      'class': 'boxplot',
+      'data-index': (d, i) => i,
+      transform: (d, i) => 'translate(' + context.cellX(i) + ',' + context.cellPrevY(i) + ')'
+    });
+
+
+    $rows_enter.append('rect').attr('class', 'shift');
+    $rows.select('rect.shift').attr(
+      {
+        'width': (d, i) => scale(q3arr[i]) - scale(q1arr[i]),
+        'height': function (d, i) {
+          var top = context.option('rowPadding', 30);
+          return (Math.max(context.rowHeight(i) - top, 10));
+        },
+        'y': 0,
+        'x': (d, i) => scale(q1arr[i])
+      });
+
+
+    $rows_enter.append('path').attr('class', 'shift');
+
+    var f = col.getWidth() / 100;
+    $rows.select('path.shift').attr('d', function (d, i) {
+
+      var left = scale(minarr[i]) * f, right = scale(maxarr[i]) * f, center = scale(medarr[i]) * f;
+      var top = context.option('rowPadding', 1);
+
+      var bottom = Math.max(context.rowHeight(i) - top, 0);
+      var middle = (bottom - top) / 2;
+
+
+      return 'M' + left + ',' + middle + 'L' + scale(q1arr[i]) + ',' + middle +
+        'M' + left + ',' + top + 'L' + left + ',' + bottom +
+        'M' + center + ',' + top + 'L' + center + ',' + bottom +
+        'M' + (scale(q1arr[i]) + scale(q3arr[i]) - scale(q1arr[i])) + ',' + middle + 'L' + (right) + ',' + middle +
+        'M' + right + ',' + top + 'L' + right + ',' + bottom;
+    });
+
+
+    context.animated($rows).attr({
+      transform: (d, i) => 'translate(' + context.cellX(i) + ',' + context.cellY(i) + ')'
+    });
+    $rows.exit().remove();
+  }
+
+}
+
+class CategoricalCellRenderer extends DefaultCellRenderer {
+  render($col: d3.Selection<any>, col: model.MyColumn, rows: any[], context: IRenderContext) {
+    var $rows = $col.datum(col).selectAll('g.categoricalcell').data(rows, context.rowKey);
+    var $rows_enter = $rows.enter().append('g').attr({
+      'class': 'categoricalcell',
+      'data-index': (d, i) => i,
+      transform: (d, i) => 'translate(' + context.cellX(i) + ',' + context.cellPrevY(i) + ')'
+    });
+
+
+    var bits = [];
+
+    var windowsize = 0;
+
+    var catindexes = [];
+
+
+    var $circle = $rows_enter.selectAll('circle').data(function (d, i) {
+      var value = col.getValue(d);
+      var data = value;
+      bits.push(data.length);
+      windowsize = (col.getWidth() / d3.max(bits));
+      return (data);
+    });
+
+
+    $circle.enter().append('circle')
+      .attr('data-index', (d, i) => i)
+      .attr('data-index', (d, i) => i)
+      .attr('cx', (d: any, i) => (i * windowsize) + (windowsize / 2))
+      .attr('cy', (d: any, i) => (context.rowHeight(i) / 2))
+      .attr('r', (windowsize / 4))
+      .attr('stroke', 'black')
+      .attr('opacity', (d) =>(d === 1) ? 1 : 0.1)
+      .attr('fill', 'black');
+
+
+    $rows_enter.append('path')
+      .attr('d', function (d, i) {
+        var value = col.getValue(d);
+        var data = value;
+        catindexes.push(data.reduce(function (b, e, i) {
+          if (e === 1) {
```

Comment:
Yes we  can do in that way also. For this part I have used random function to generate data. Because I cannot make category for the actual data. This is just the dummy. I will  work on the another possible way  also then.